lib/fog/vsphere/requests/compute/create_vm.rb

```diff
@@ -180,19 +223,22 @@ def controller_get_shared_from_options options
           end
         end
 
-        def create_disk disk, index = 0, operation = :add, controller_key = 1000
+        def create_disk disk, index = 0, operation = :add, controller_key = 1000, storage_pod_selected = nil
           if (index > 6) then
             _index = index + 1
           else
             _index = index
           end
+          # If we deploy the vm on a storage pool, datastore has to be an empty string
+          datastore = ''
+          datastore = "[#{disk.datastore}]" unless storage_pod_selected
```

Comment:
I think this would read better if you did an `if ... else` like:

```ruby
if storage_pod_selected
  datastore = ''
else
  datastore = "[#{disk.datastore}]"
end
```